api/v1beta1/foundationdbcluster_types.go

```diff
@@ -1563,23 +1567,28 @@ type ProcessAddress struct {
 // representation.
 func ParseProcessAddress(address string) (ProcessAddress, error) {
 	result := ProcessAddress{}
-	components := strings.Split(address, ":")
 
-	if len(components) < 2 {
+	ipEnd := strings.Index(address, "]:") + 1
```

Comment:
Instead of building this custom logic I would prefer to use `net.SplitHostPort` that should make everything more robust.

Reply:
How would that work with the custom flags in our addresses like the `:tls` suffix?

Reply:
I got a solution for that in mind (more or less the same FDB currently does). I would work on that in https://github.com/FoundationDB/fdb-kubernetes-operator/issues/352 that should simplify thee logic here. For the PR itself that's fine and we can improve things step by step.